src/test/java/org/junitpioneer/testkit/PioneerAssertContainer.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.testkit;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import org.junit.platform.engine.TestExecutionResult;
+
+public class PioneerAssertContainer {
+
+	public static class PioneerAssert extends AbstractAssert<PioneerAssert, ExecutionResults>
+			implements SingleReportEntryAssert, FailureAssert, ExceptionAssert {
+
+		public PioneerAssert(ExecutionResults actual) {
+			super(actual, PioneerAssert.class);
+		}
+
+		public static PioneerAssert assertThat(ExecutionResults actual) {
+			return new PioneerAssert(actual);
+		}
+
+		public SingleReportEntryAssert hasSingleReportEntry() {
+			isNotNull();
+
+			List<Map<String, String>> reportEntries = actual.reportEntries();
+			Assertions.assertThat(reportEntries).hasSize(1);
+			Map<String, String> reportEntry = reportEntries.get(0);
+			Assertions.assertThat(reportEntry).hasSize(1);
+
+			return this;
+		}
+
+		public FailureAssert hasSingleFailedTest() {
```

Comment:
i might suggest to have one basic method like `hasNumberOfFailedTests(int number)` and than just use that within this method. as i think this assert might be usable for other cases in the future too.

Reply:
I would like to keep the single methods and offer additional methods where you can pass the number, because we have plenty of tests where we expect exactly one. So why force the writer to pass magic numbers and make it less readable in those cases? Don't get me wrong. We need the basic methods for dynamic parameters, but this PR shows that the single ones makes sense too.

Reply:
i never said to skip that method, i said we should extract its functionality into an generic one, which we use within this method